### PR TITLE
[trainer, algo, cfg] feat: add teacher-anchored continuous distillation losses for diffusion OPD

### PR DESCRIPTION
**Pull request by cr-gao**
### What does this PR do?

Implements the "Distillation Loss/Reward kernels" item of RFC #293: two continuous-latent distillation losses registered in `DIFFUSION_LOSS_REGISTRY`, usable standalone via `diffusion_loss.loss_mode` or as an auxiliary term alongside a policy-gradient loss via `use_distill_loss` / `distill_loss_coef` (mirrors `use_kl_loss`).

- `distill_kl`: Gaussian transition-kernel KL between student and teacher reverse-SDE means (same form as `KLLoss`, anchored to `teacher_prev_sample_mean`).
- `distill_fm_mse`: MSE between student and teacher flow/noise predictions (`teacher_noise_pred`), bypassing log-prob / Top-K machinery entirely.

Note on RFC #293 3: extending verl's `DISTILLATION_LOSS_REGISTRY` does not fit as written — `DistillationLossSettings` hard-requires top-k xor estimator (token) semantics, and the diffusion actor never passes through verl's token-based distillation pipeline. verl-omni's own `DIFFUSION_LOSS_REGISTRY` is the natural extension point. This PR is scoped to compose with (not replace) the TeacherWorker / trajectory-caching / pipeline-integration workstream: teacher targets enter through the documented `teacher_*` batch keys, per-step sliced in both engine impls (mirrors `ref_*`). Producing those tensors is out of scope here.

### Checklist Before Starting

- [x] Search for similar PRs: https://github.com/verl-project/verl-omni/pulls?q=is%3Apr+distillation (no PRs touch distillation; #293 has no linked PRs)
- [x] Format the PR title as `[{modules}] {type}: {description}`

### Test

CPU unit tests (new file `tests/trainer/diffusion/test_diffusion_distillation_losses_on_cpu.py`, 20 cases):

```bash
pytest tests/trainer/diffusion/test_diffusion_distillation_losses_on_cpu.py  # 20 passed
pytest tests/trainer/                                                        # 116 passed
pre-commit run --all-files                                                    # clean
```

Coverage: closed-form KL/MSE value checks, zero-at-identity, monotonicity, image/video shapes, `validate_inputs` missing-key errors, registry/config plumbing, and signal-integrity of the auxiliary combination (`total == pg_loss + coef * distill_loss`; default-off leaves existing losses bit-identical).

The engine-side `teacher_*` per-step slicing follows the existing `ref_*` pattern and is a guarded no-op unless teacher tensors are present in the batch; it becomes exercisable end-to-end once a teacher producer lands (follow-up scope of #293).

### API and Usage Example

```yaml
# auxiliary distillation term alongside a policy-gradient loss
actor_rollout_ref.actor.use_distill_loss: true
actor_rollout_ref.actor.distill_loss_mode: distill_kl   # or distill_fm_mse
actor_rollout_ref.actor.distill_loss_coef: 1.0

# or pure supervised distillation
actor_rollout_ref.actor.diffusion_loss.loss_mode: distill_kl
```

Batch-key contract (produced by the future TeacherWorker, evaluated at the student's explored latents):

- `distill_kl`: `teacher_prev_sample_mean` (bs, num_steps, s, c), per-step sliced like `ref_prev_sample_mean`
- `distill_fm_mse`: `teacher_noise_pred`, handled like `ref_noise_pred`

### Design & Code Changes

- `verl_omni/trainer/diffusion/diffusion_algos.py`: `DistillKLLoss`, `DistillFlowMatchingMSELoss` (registered; follows Step 3 of the diffusion algorithm integration guide — adapter/estimator/launch-script checklist items do not apply since this adds loss kernels, not a new algorithm)
- `verl_omni/workers/config/diffusion/actor.py`: new modes in `valid_modes`; `use_distill_loss` / `distill_loss_mode` / `distill_loss_coef` fields + validation
- `verl_omni/workers/utils/losses.py`: auxiliary distillation term in `diffusion_loss()`, mirroring the `use_kl_loss` block
- `verl_omni/workers/engine/fsdp/diffusers_impl.py`, `verl_omni/workers/engine/veomni/diffusion_impl.py`: `teacher_*` batch-key slicing, mirroring `ref_*`
- `verl_omni/trainer/config/diffusion/actor/diffusion_actor.yaml` + regenerated `_generated_*.yaml`

### CODE REVIEW

**Comment by zhtmike** [comment]
well, a better way is to draft a rough PR verifying that the idea is working, then split the PR like loss, pipeline, etc.
Otherwise we don't know if the loss is correct.

**Comment by cr-gao** [comment]
> well, a better way is to draft a rough PR verifying that the idea is working, then split the PR like loss, pipeline, etc. Otherwise we don't know if the loss is correct.

Thanks for the suggestion.
Ran the rough end-to-end verification with SD3.5-medium OCR on 3 GPUs. Teacher = FlowGRPO LoRA run (val 0.21- 0.24 base -> ~0.52), and student = fresh model trained with pure distill_kl against the merged teacher. There is no reward signal in the loss. distill_kl fell 4.5e-4 -> 1.3e-5 and student val reached teacher level by step 20 (0.65 peak, 0.55–0.59 settled, vs teacher 0.50–0.57 across evals). Teacher targets come from a ref-worker checkpoint override, wiring + LoRA-merge script + curves in draft #304.

**Comment by SamitHuang** [comment]
> > well, a better way is to draft a rough PR verifying that the idea is working, then split the PR like loss, pipeline, etc. Otherwise we don't know if the loss is correct.
> 
> Thanks for the suggestion. Ran the rough end-to-end verification with SD3.5-medium OCR on 3 GPUs. Teacher = FlowGRPO LoRA run (val 0.21- 0.24 base -> ~0.52), and student = fresh model trained with pure distill_kl against the merged teacher. There is no reward signal in the loss. distill_kl fell 4.5e-4 -> 1.3e-5 and student val reached teacher level by step 20 (0.65 peak, 0.55–0.59 settled, vs teacher 0.50–0.57 across evals). Teacher targets come from a ref-worker checkpoint override, wiring + LoRA-merge script + curves in draft #304.

The training results LGTM. @zhtmike PTAL

btw, how many PR splits are needed toward the e2e OPD support? @cr-gao

**Review by zhtmike — APPROVED** [review]
Clear PR! LGTM

**Comment by cr-gao** [comment]
@SamitHuang 3 more PRs after this one to reach the RFC targets.

1. This PR: distill loss kernels and the `teacher_*` consumer contract. I will
   push a small validation guard for `distill_fm_mse` on the policy-gradient
   path per the inline review before merge.
2. Teacher runtime MVP: independent `Role.TeacherModel`, frozen teacher worker,
   single-teacher manager, colocated in `global_pool`, one training step wired
   e2e. Supersedes teacher-in-ref; #304 stays a draft as the verified baseline.
3. Standalone `teacher_pool`: decoupled teacher resources plus ref and teacher
   coexistence validation.
4. Recipe: LoRA-merge script, SD3.5-Medium OCR example, docs.

**Review by SamitHuang — APPROVED** [review]
LGTM
